robotframework-ls/src/robotframework_ls/robotframework_ls_impl.py

```diff
@@ -540,6 +546,44 @@ def func(monitor):
         log.info("Unable to get RF info (no api available).")
         return None
 
+    @command_dispatcher(ROBOT_OPEN_FLOW_EXPLORER_INTERNAL)
+    def _open_flow_explorer(self, *arguments):
+        current_doc_uri = arguments[0]["currentFileUri"]
+        html_bundle_flow_folder_path = arguments[0]["htmlBundleFolderPath"]
+        html_bundle_flow_file_path = os.path.join(
+            html_bundle_flow_folder_path, _FLOW_EXPLORER_BUNDLE_HTML_FILE_NAME
+        )
+        tmp_bundle_flow_file_path = os.path.join(
+            html_bundle_flow_folder_path, "tmp_" + _FLOW_EXPLORER_BUNDLE_HTML_FILE_NAME
```

Comment:
Using the `tempfile` is tricky. 
We need the file to open in the browser, but using a temporary file as in (for example) a `NamedTemporaryFile` will delete the file when the context is closed and frontend cannot open the inexistent file. 
To add to the problem, using temporary files but not deleting them in the app, will pile up a series of temporary files. Is this OK to happen?
If there is something that I've missed please let me know. 

I'm implementing a version using temporary files, but without deleting them.

Reply:
I agree with you: the preferred approach would probably be:
1. Get temporary dir
2. Create a directory in the temporary dir (it can always be the same name)
3. Erase old files so that no more than 5 files are in the directory (based on the expected pattern)
4. Create a file with the expected pattern (maybe something as `0001_tmp_flow.html`) -- the `0001` would always be equal to 1 + the latest value we had already written.
5. Provide that file

We don't have such an utility right now, but given that this doesn't seem something uncommon we should probably make this a utility somewhere such as `robocorp_ls_core.rotating_file.py` (with proper tests).